File: Frontend/vitanova/src/app/FrontOffice/Models/post.ts
import { Comment } from "./comment";
import { FileEntity } from "./file-entity";

export class Post {
    
        id!: string;
        content!: string;
        postedBy!: string;
        date!: Date;
        likeCount!: number;
        viewCount!: number;
        tags!: string[];
        images!:FileEntity[] ;
        comments!: Comment[];

    

}

File: Backend/src/main/java/com/example/backend/Entities/Post.java
package com.example.backend.Entities;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.AccessType;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;
import java.util.Set;

@Document
@NoArgsConstructor
@AllArgsConstructor
@Data
@FieldDefaults(level = AccessLevel.PRIVATE)

public class Post  {

    @Id
    String id;
    String content;
    String postedBy;
    Date date;
    Integer likeCount;
    Integer viewCount;
    List<String> tags;
    Set<FileEntity>images;

    @DBRef
    Set<Comment>comments;




}
